drivers/regulator/regulator_tps55287.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * SPDX-FileCopyrightText: Copyright The Zephyr Project Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_tps55287
+
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/linear_range.h>
+#include <zephyr/sys/util.h>
+
+LOG_MODULE_REGISTER(tps55287, CONFIG_REGULATOR_LOG_LEVEL);
+
+#define TPS55287_REG_REF     0x00U
+#define TPS55287_REG_VOUT_FS 0x04U
+#define TPS55287_REG_MODE    0x06U
+
+#define TPS55287_REG_MODE_OE BIT(7)
+
+static const struct linear_range core_ranges[] = {
+	LINEAR_RANGE_INIT(800000u, 2500u, 0xf0, BIT_MASK(11)),
+	LINEAR_RANGE_INIT(800000u, 5000u, 0x50, BIT_MASK(11)),
+	LINEAR_RANGE_INIT(800000u, 7500u, 0x1a, BIT_MASK(11)),
+	LINEAR_RANGE_INIT(800000u, 10000u, 0x00, BIT_MASK(11)),
+};
+
+struct regulator_tps55287_config {
+	struct regulator_common_config common;
+	struct i2c_dt_spec i2c;
+};
+
+struct regulator_tps55287_data {
+	struct regulator_common_data data;
+};
+
+static unsigned int regulator_tps55287_count_voltages(const struct device *dev)
+{
+	return linear_range_group_values_count(core_ranges, ARRAY_SIZE(core_ranges));
+}
+
+static int regulator_tps55287_list_voltage(const struct device *dev, unsigned int idx,
+					   int32_t *volt_uv)
+{
+	for (uint8_t i = 0U; i < ARRAY_SIZE(core_ranges); i++) {
+		if (linear_range_get_value(&core_ranges[i], idx, volt_uv) == 0) {
+			return 0;
+		}
+		idx -= linear_range_values_count(&core_ranges[i]);
+	}
+
+	return -EINVAL;
+}
+
+static int regulator_tps55287_set_voltage(const struct device *dev, int32_t min_uv, int32_t max_uv)
+{
+	const struct regulator_tps55287_config *config = dev->config;
+	uint8_t buf[3] = {0};
+	uint16_t idx;
+	uint8_t vout_fs_reg;
+	int ret;
+
+	ret = i2c_reg_read_byte_dt(&config->i2c, TPS55287_REG_VOUT_FS, &vout_fs_reg);
+	if (ret < 0) {
+		return ret;
+	}
+
+	vout_fs_reg &= 0x03;
```

Comment:
Could the bitmask used here be defined with a name, so that the code is easier to follow?